libcloud/compute/drivers/outscale.py

```diff
@@ -7948,3 +7949,283 @@ def _to_key_pair(self, key_pair):
+
+        :return: a dict if error else the ciphertext.
+        :rtype: ``dict`` or ``str``
+        """
+        action = "EncryptPlaintext"
+        data = {"MasterKeyId" : master_key_id, "Plaintext" : base64.b64encode(bytes(plaintext, 'utf-8')).decode('ascii'), "EncryptionContext" : encryption_context, "DryRun": dry_run}
+        response = self._call_api(action, json.dumps(data))
+        if response.status_code == 200:
+            return response.json()['Ciphertext']
+        return response.json()
+
+
+    def ex_decrypt_ciphertext(
+        self,
+        ciphertext: str,
+        encryption_context: dict = dict(),
+        dry_run: bool = False,
+    ):
+        """
+        Encrypt a plain key using a CMK.
+
+        :param      ciphertext: This is text you decrypting.
+        :type       ciphertext: ``str``
+
+        :param      encryption_context: This is the context you are using.
+        :type       encryption_context: ``dict``
+
+        :param      dry_run: If true, checks whether you have the required
+        permissions to perform the action.
+        :type       dry_run: ``bool``
+
+        :return: a dict if error else the plaintext.
+        :rtype: ``dict`` or ``str``
+        """
+        action = "DecryptCiphertext"
+        data = {"Ciphertext" : ciphertext, "EncryptionContext" : encryption_context, "DryRun": dry_run}
+        response = self._call_api(action, json.dumps(data))
+        if response.status_code == 200:
+            return base64.b64decode(response.json()['Plaintext']).decode('ascii')
+        return response.json()
+
+    def ex_generate_data_key(
+        self,
+        master_key_id : str,
+        size : int,
+        generate_plaintext: bool,
+        encryption_context: dict = dict(),
+        dry_run: bool = False,
+    ):
+        """
+        Generates a data key using a master key.
+
+        :param      master_key_id: This is the master_key_id we want to use.
+        :type       master_key_id: ``str``
+
+        :param      size: Size of the data between 1 and 1024.
+        :type       size: ``int``
+
+        :param      generated_plaintext: If true, data received won't be encrypted.
+        :type       generated_plaintext: ``bool``
+
+        :param      encryption_context: This is the context you are using.
+        :type       encryption_context: ``dict``
+
+        :param      dry_run: If true, checks whether you have the required
+        permissions to perform the action.
+        :type       dry_run: ``bool``
+
+        :return: a dict containing 'Plaintext' if generate_plaintext is true and `Ciphertext`.
+        :rtype: ``dict`` of ``str``
+        """
+        action = "GenerateDataKey"
+        data = {"MasterKeyId" : master_key_id, "Size": size, "GeneratePlaintext" : generate_plaintext,  "EncryptionContext" : encryption_context, "DryRun": dry_run}
+        response = self._call_api(action, json.dumps(data))
+        return response.json()
+
+    def ex_read_master_keys(
```

Comment:
I think it would be better to call this method `ex_list_master_keys` as all other read methods are called that way